### PR TITLE
Aggregation improvements

### PR DESCRIPTION
**Pull request by jgallagher59701**
The aggregation_improvements branch includes code in the dap module to free Array variable memory once those variables have been serialized. This reduces the memory required by the BES when serializing responses with several variables. Note that the code calls clear_local_data() for every variable, but it's primarily Arrays where the benefit is seen.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -82,21 +83,22 @@ install:
 #
 # We want the hdf5 handler build to find the hdf5 lib headers from the
 # source tarball we just built so run configure using CPPFLAGS=...
+# removed CPPFLAGS=-I$HOME/deps/include added --with-hdf5... jhrg 9/3/15
 
 before_script:
   - ln -s configure_modules.ac configure.ac
   - autoreconf --force --install --verbose
-  - ./configure --prefix=$HOME/build CPPFLAGS=-I$HOME/deps/include
+  - ./configure --prefix=$HOME/build --with-hdf5=$HOME/deps
```
Comment (mrpatrickwest) [inline .travis.yml]: Why is hdf5 required to build the BES? I would think that the only dependency would be libdap?

File: install_bes_source_deps.sh [hunk 2]
```diff
@@ -2,18 +2,31 @@
```
Comment (mrpatrickwest) [inline install_bes_source_deps.sh]: Is this file part of the travis-ci build? There's no comments saying what this file is for.

File: modules/ncml_module [hunk 3]
```diff
@@ -1 +1 @@
-Subproject commit 92e95489cdf742ef807a753a09f858a6b44c87e5
```
Comment (mrpatrickwest) [inline modules/ncml_module]: Same question as above, curious why ncml_module is a dependency, submodule in this case, for the BES? Same for xml_data_handler

File: timing/mkAllTests [hunk 4]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
```
Comment (mrpatrickwest) [inline timing/mkAllTests]: Is this something run by travis? Again, no comments explaining what this file is for.

File: timing/mkCurlTests [hunk 5]
```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
```
Comment (mrpatrickwest) [inline timing/mkCurlTests]: Same comment as above for mkAllTests

**Comment by mrpatrickwest** [comment]
where's the code for this branch. Comment says that there's a code fix in the dap module to free memory, but I don't see source code committed for the pull request.

File: install_bes_source_deps.sh (hunk 2 above)
Comment (jgallagher59701) [inline install_bes_source_deps.sh]: It is used to install bes source dependencies for the travis ci build

File: timing/mkAllTests (hunk 4 above)
Comment (jgallagher59701) [inline timing/mkAllTests]: No, this is not run by travis. All the files in 'timing' are for running timing analysis on the BES.

File: modules/ncml_module (hunk 3 above)
Comment (jgallagher59701) [inline modules/ncml_module]: Because the travis build runs the 'modules' build of the BES.

**Pull request merged**